core/src/main/java/org/apache/accumulo/core/client/admin/TabletInformation.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.accumulo.core.client.admin;
+
+import java.util.Objects;
+
+import org.apache.accumulo.core.data.TableId;
+import org.apache.accumulo.core.util.NumUtil;
+
+public class TabletInformation {
+
+  private final String tableName;
+  private final String endRow;
+  private final String prevEndRow;
+  private final int numFiles;
+  private final int numWalLogs;
+  private final long numEntries;
+  private final long size;
+  private final String status;
+  private final String location;
+  private final String dir;
+  private final TableId tableId;
+  private final TabletHostingGoal goal;
+
+  public TabletInformation(String tableName, TableId tableId, String endRow, String prevEndRow,
+      int numFiles, int numWalLogs, long numEntries, long size, String status, String location,
+      String dir, TabletHostingGoal goal) {
+    this.tableName = tableName;
+    this.tableId = tableId;
+    this.endRow = endRow;
+    this.prevEndRow = prevEndRow;
+    this.numFiles = numFiles;
+    this.numWalLogs = numWalLogs;
+    this.numEntries = numEntries;
+    this.size = size;
+    this.status = status;
+    this.location = location;
+    this.dir = dir;
+    this.goal = goal;
+  }
```

Comment:
If its workable it would be nice to make this object a wrapper around TabletMetadata, then it does not need to duplicate alot of the fields.

```suggestion
  private final TabletMetadata tabletMetadata;

  public TabletInformation(TabletMetadata tabletMetadata) {
    this.tabletMetadata = tabletMetadata;
  }
```

Reply:
@keith-turner The issue I run into when attempting this is that it will not compile due to TabletMetadata being a non-public API reference in a public class. The apilyzer maven check prevents the build. Do you know of an alternative means where the same objective could be met? I had run into this earlier when attempting to use KeyExtent. I had to re-work to avoid that same issue as well.

Reply:
Oh interesting. One way around this is the following :

- Create a TabletInformation interface w/ all the getter methods in this class.
- Create a non public API class called TabletInformationImpl that wraps TabletMetadata. Internal code creates TabletInformationImpl and returns them when TabletInformation type is needed.

This is a similar strategy to TabletId and TabletIdImpl which wraps KeyExtent.

Reply:
Thanks. I'll give that a shot. I definitely prefer not having to use all of the constructor fields.

Reply:
@keith-turner the onc parameter I couldn't figure out how to determine within the class itself was the TabletState value, so I pass it in with the TabletMetadata information in the constructor. If you have a suggestion as to how to avoid this please let me know.

Reply:
Passing any additional information to the constructor if it can not be obtained from tabletmetadata seems fine.